File: typescript/CDO/ActionResponse.ts
var xmpp = require('../node-xmpp/packages/client').xmpp;
var xml = require('../node-xmpp/packages/xml');
var Guid = require('../Misc/Guid');
import {CDO} from './CDO';

export class ActionResponse
{
    public ID:string;
    public Successful:boolean;
    public TimedOut:boolean = false;
    /*
    public List<string> InformationMessages = new List<string>();
    public List<string> WarningMessages = new List<string>();
    public List<string> ErrorMessages= new List<string>();*/
    public Payload:any;

    public constructor(Element:any)
    {
        if(!Element)
            return;
        this.ID = Element.attrs.id as string;

        this.Payload = Element.children[0];
        this.Successful = CDO.GetBoolean(this.Payload, "successful");
    }

    public GetXML(ReturnTypename : string, ActionName : string) : any
    {
        let RespEl = xml("actionresponse", {"xmlns": CDO.Namespace});
        RespEl.attrs.id = this.ID;
        RespEl.attrs.name =  ActionName;
        let GenRespEl =  xml(ReturnTypename, {/*"xmlns": CDO.Namespace*/});
        CDO.AddBoolean(GenRespEl, "successful", this.Successful);
        RespEl.append(GenRespEl);
        return RespEl;
    }
}

File: NET/CDO/ActionResponse.cs
using System;
using System.Collections.Generic;
using System.Xml.Linq;
using System.Linq;

namespace CDO
{
	public class ActionResponse
	{
		public string ID;
		public bool Successful;
		public bool TimedOut = false;
		/*
		public List<string> InformationMessages = new List<string>();
		public List<string> WarningMessages = new List<string>();
		public List<string> ErrorMessages= new List<string>();*/
		public XElement Payload;

		public ActionResponse(XElement Element)
		{
			ID = Element.Attribute("id").Value;

			Payload = Element.Elements().First();
			Successful = CDO.GetBoolean(Payload, "successful");
		}

		public ActionResponse()
		{
		}

		public XElement GetXML(string ReturnTypename, string ActionName)
		{
			XElement RespEl = new XElement(CDO.Namespace + "actionresponse");
			RespEl.SetAttributeValue("id", ID);
			RespEl.SetAttributeValue("name", ActionName);
			XElement GenRespEl = new XElement(CDO.Namespace + ReturnTypename);
			CDO.SetBoolean(GenRespEl, "successful", Successful);
			RespEl.Add(GenRespEl);
			return RespEl;
		}
	}
}
